contracts/anchors/bridged/AnchorPoseidon2.sol

```diff
@@ -120,28 +122,24 @@ abstract contract AnchorPoseidon2 is MerkleTreePoseidon, ReentrancyGuard {
   ) external payable nonReentrant {
     require(_fee <= denomination, "Fee exceeds transfer value");
     require(!nullifierHashes[_nullifierHash], "The note has been already spent");
-    require(isKnownRoot(_root), "Cannot find your merkle root"); // Make sure to use a recent one
+    require(isKnownRoot(_roots[0]), "Cannot find your merkle root");
+		require(_roots.length >= edgeList.length + 1, "Incorrect root array length");
```

Comment:
spacing is off

Reply:
also we should parse the new variable as we discussed:
```
bytes32[2] roots = abi.decode(_roots, (bytes32[2]));
```
follow previous examples that we have for parsing the proof to do it correctly.